pkgs/development/libraries/boost/generic.nix

```diff
@@ -40,12 +41,19 @@ let
   # To avoid library name collisions
   layout = if taggedLayout then "tagged" else "system";
 
+  libcxxflags = if enableLibcxx then
+                  "-nostdinc++ -I${llvmPackages.libcxx}/include/c++/v1"
+                else
+                  "";
+
   cflags = if enablePIC && enableExceptions then
```

Comment:
Actually, testing something else: using stdenv=libcxxStdenv might be a better alternative...